src/cmd-generate-hashlist

```diff
@@ -0,0 +1,199 @@
+        :param generator: Optional generator override
+        :type generator: str
+        :param timestamp: Optional timestamp override
+        :type timestamp: str
+        :raises: KeyError
+        """
+        super()
+        self._metadata = metadata
+        if release is None:
+            release = self._metadata['buildid']
+        self['release'] = release
+
+        self['meta'] = {}
+        if generator is None:
+            generator = 'coreos-assembler-' + self._metadata['coreos-assembler.container-image-git']['commit']  # noqa
+        self['meta']['generator'] = generator
+        if timestamp is None:
+            timestamp = datetime.datetime.utcnow().isoformat()
+        self['meta']['timestamp'] = timestamp
+        self['meta']['image'] = os.path.sep.join([
+            'builds', self._metadata['ostree-version'],
+            self._metadata['coreos-assembler.basearch'],
+            self._metadata['images']['qemu']['path']])
+
+        self['hashes'] = {}
+
+    def populate_hash_list(self):
+        """
+        Populates the hashlist instance with ostree and initramfs hashes.
+
+        :param checksum: The ostree checksum from the metadata
+        :type checksum: str
+        :param commit: The ostree commit
+        :type commit: str
+        :param commit_tar: Path to the ostree tar from the metadata
+        :type commit_tar: str
+        :param hashlist: Initialized hash list instance
+        :type hashlist: HashListV1
+        :returns: Nothing
+        :rtype: None
+        :raises: IndexError
+        """
+        checkout = 'tmp/repo/tmp/keylime-checkout'
+        commit_tar = os.path.join(
+            self._metadata.build_dir,
+            self._metadata['images']['ostree']['path'])
+
+        import_ostree_commit(
+            'tmp/repo',
+            self._metadata['ostree-commit'],
+            commit_tar,
+            force=True)
+        subprocess.check_call([
+            'ostree', 'checkout',
+            '--repo=tmp/repo', '-U',
+            self._metadata['ostree-commit'], checkout])
+        self.hash_from_path(checkout)
+
+        # Extract initramfs contents
+        initramfs_path = glob.glob(
+            os.path.join(
+                checkout, 'usr/lib/modules/*/initramfs.img'))[0]
+        initramfs_path = os.path.realpath(initramfs_path)
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # TODO: This pipeline isn't working as expected...
+            skipcpio = subprocess.Popen(
+                ['/usr/lib/dracut/skipcpio', initramfs_path],
+                stdout=subprocess.PIPE,
+                cwd=tmpdir)
+            gunzip = subprocess.Popen(
+                ['gunzip', '-c'],
+                stdin=skipcpio.stdout,
+                stdout=subprocess.PIPE,
+                cwd=tmpdir)
+            skipcpio.stdout.close()
+            cpio = subprocess.Popen(
+                ['cpio', '-idmv'],
+                stdin=gunzip.stdout,
+                cwd=tmpdir)
```

Comment:
Minor: I think only this `cwd=tmpdir` is strictly necessary. The other ones should be fine without since they don't make use of the working directory.

Reply:
You're right. Easy update :smile: